.github/workflows/ci.yml

```
@@ -0,0 +1,41 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - openssl1.1
+  pull_request:
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        compiler: ["g++", "clang++"]
+        build_type: ["", "ASAN"]
+    steps:
+      - name: "Checkout"
+        uses: actions/checkout@v2
+
+      - name: "Load Docker cache"
+        uses: satackey/action-docker-layer-caching@v0.0.11
+
+      - name: "Build Docker image"
+        run: |
+          docker build -t buildenv - < docker-build-env/Dockerfile
+
+      - name: "Build MoCOCrW"
+        run: |
+          mkdir build
+          docker run --rm -u $UID -v $PWD:/src:rw -w /src/build buildenv bash -c \
```

My endless shellscript paranoia would prefer:

```suggestion
          docker run --rm -u "$UID" -v "$PWD:/src:rw" -w /src/build buildenv bash -c \
```

---

:+1: It's not paranoia if they're really after you.